### PR TITLE
Linode changes

### PR DESCRIPTION
**Pull request by InTheCloudDan**
##### ISSUE TYPE

- Feature Pull Request
##### ANSIBLE VERSION

```
ansible 2.1.0.0
```
##### SUMMARY

Added options when creating a new Linode. Everything exposed via https://www.linode.com/api/linode/linode.update is now able to be used in an Ansible task.

```

```

### CODE REVIEW

File: cloud/linode/linode.py [hunk 1]
```diff
@@ -37,12 +37,77 @@
      - To keep sanity on the Linode Web Console, name is prepended with LinodeID_
     default: null
     type: string
+  displaygroup:
+    description:
+     - Add the instance to a Display Group in Linode Manager (alphanumeric, dashes, underscore)
+    default: null
```
Comment (gundalow) [inline cloud/linode/linode.py:43]: Please add the following at the same indent level as `description`
`version_added: 2.2`

File: cloud/linode/linode.py [hunk 2]
```diff
@@ -445,16 +537,32 @@ def main():
                                                      'restarted']),
             api_key = dict(),
             name = dict(type='str'),
+            alert_bwin_enabled = dict(type='int', default=1),
```
Comment (gundalow) [inline cloud/linode/linode.py]: I think some of these new options should be `type='bool'`

File: cloud/linode/linode.py [hunk 3]
```diff
@@ -445,16 +537,32 @@ def main():
...
+            alert_bwin_threshold = dict(type='int'),
+            alert_bwout_enabled = dict(type='int', default=1),
```
Comment (gundalow) [inline cloud/linode/linode.py]: I think some of these new options should be `type='bool'`

**Comment by gundalow** [comment]
needs_revision

**Comment by gregdek** [comment]
Thanks @InTheCloudDan for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by InTheCloudDan** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @InTheCloudDan. To the current maintainers, @zbal please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks @InTheCloudDan for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

For help on how to do this cleanly please see http://docs.ansible.com/ansible/community.html#contributing-code-features-or-bugfixes

[This message brought to you by your friendly Ansibull-bot.]

**Comment by InTheCloudDan** [comment]
ready_for_review

File: cloud/linode/linode.py [hunk 4]
```diff
@@ -37,12 +37,95 @@
...
+    version_added: "2.2"
   linode_id:
     description:
      - Unique ID of a linode server
     aliases: [ 'lid' ]
     default: null
     type: integer
+  additional_disks:
+    description:
+     - > List of dictionaries for creating additional disks that are added to the Linode configuration settings. Dictionary uses: Size, Label, Type. Size is in MB.
```
Comment (gundalow) [inline cloud/linode/linode.py:54]: Please remove the `>` character

File: cloud/linode/linode.py [hunk 5]
```diff
@@ -37,12 +37,95 @@
      - To keep sanity on the Linode Web Console, name is prepended with LinodeID_
     default: null
     type: string
+  displaygroup:
+    description:
+     - Add the instance to a Display Group in Linode Manager
+    default: null
+    type: string
+    version_added: "2.2"
```
Comment (gundalow) [inline cloud/linode/linode.py:45]: Ansible 2.2 has been branched already so this and the other instances for parameters you've added need chanming to `2.3`.

File: cloud/linode/linode.py [hunk 6]
```diff
@@ -37,12 +37,95 @@
...
+  alert_bwout_threshold:
+    description:
+    - Set threshold in MB of bandwidth out alerts.
+    default: null
+    type: integer
+    version_added: "2.2"
+  alert_bwquota_enabled:
+    description:
+    - Set status of bandwidth quota alerts as percentage of network tranfer quota.
```
Comment (gundalow) [inline cloud/linode/linode.py:83]: s/tranfer/transfer/

**Review by gundalow — COMMENTED** [review]
needs_revision

Also needs rebasing at this file apparently has merge conflicts.

**Pull request closed without merging**